pkg/kubelet/status/manager.go

```diff
@@ -182,16 +186,37 @@ func (m *manager) TerminatePods(pods []*api.Pod) bool {
 				Terminated: &api.ContainerStateTerminated{},
 			}
 		}
-		select {
-		case m.podStatusChannel <- podStatusSyncRequest{pod, pod.Status}:
-		default:
+		newStatus := m.updateStatusInternal(pod, pod.Status)
```

Comment:
Note to reviewers: This is a slight change in behavior, as we weren't caching the termination status before. I think this is the more correct way, although it does not enforce the same invariants as `SetPodStatus` (maintain `StartTime` and `LastTransitionTime`) - but I don't think that is necessary here.